src/installation/guides/zfs.md

```diff
@@ -0,0 +1,227 @@
+# ZFS On Root
+
+This installation guide assumes that an already existing Full Void Linux install
+is available. Not just a Live USB.
+
+If an existing Void install does not exist please consult one of the [other
+installation guides](./index.md) or use a custom iso such as
+[hrmpf](https://github.com/leahneukirchen/hrmpf) then continue along with this
+guide.
+
+## Setup
+
+From the permanent system, install the `zfs` package.
+
+Then ensure the ZFS module is loaded with
+[modprobe(1)](https://man.voidlinux.org/modprobe):
+
+`# modprobe zfs`
+
+Locate the disk to format using [fdisk(8)](https://man.voidlinux.org/fdisk.8) or
+[lsblk(8)](https://man.voidlinux.org/lsblk.8):
+
+`# fdisk -l`
+
+## Partitioning
+
+Prepare the drive for installation by creating one of the following partition
+schemes using [cfdisk(8)](https://man.voidlinux.org/cfdisk.8) or another
+partitioning tool:
+
+`# cfdisk -z /dev/sda`
+
+**Warning:** The disk being partitioned will be formatted and any existing data
+on the disk will be destroyed.
+
+For a BIOS/MBR system:
+
+```
+Partition    Size       Type
+1             ?G     Solaris Root(bf00)
+```
+
+For a BIOS/GPT system:
+
+```
+Partition    Size       Type
+1             2M     BIOS boot    
+2             ?G     Solaris Root
+```
+
+ZFS pools should use id's to maintain universal functionality across computers
+and to prevent the misrecognition of drives.
+
+Id's can be located in the `/dev/disk/by-id` folder:
+
+Note: `wwn-` entries and `ata-` entries are referencing equivalent drives and
+either can be used.
+
+```
+$ ls -l /dev/disk/by-id
+total 0
+lrwxrwxrwx 1 root root 10 Apr 16 17:34 ata-Hitachi_HDS721010CLA332_JP6911HD2AHS9F-part1 -> ../../sda1
+lrwxrwxrwx 1 root root 10 Apr 16 17:34 wwn-0x5000cca373e0f5d9-part1 -> ../../sda1
+```
+
+Create a pool specifying the previously created Solaris Root Partition from the
+disk:
+
+Note: Many users with Advanced Format hard drives will want to also add the `-o
+ahisft=12` argument. This sets the drive's logical sectors to 4k in size.
+Additionally many may want to utilize compression with their pools using either
+lz4 or another alogorithm, see
+[zfs(8)](https://man.voidlinux.org/zfs.8#Native_Properties). This can be done
+after the creation of the pool using: `# zfs set compression lz4 zroot`
```

Comment:
I'd mention that the user should look into the man page for all the relevant filesystem options, and that most of them can also be set afterwards. No need for an example command like `zfs set compression...`.

I also think this paragraph shouldn't be a Note at all.

Perhaps we should do something like this guide https://github.com/zdykstra/zfsbootmenu/wiki#zfs-pool-creation to give a brief example of flags that they may want to set, all of which should be looked into. The only ones that need to be explained are the ones that you already included, I think.